src/main/java/Storage.java

```diff
@@ -0,0 +1,130 @@
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+/**
+ * Represents the storage class, which is responsible for
+ * loading and saving tasks into the save file.
+ */
+public class Storage {
+    private Path saveFilePath;
+
+    /**
+     * Creates a Storage object, which is used to load and save tasks
+     * into a save file.
+     *
+     * @param fileName the name of the save file
+     */
+    public Storage(String fileName) {
+        String home = System.getProperty("user.home");
+
+        Path dirPath = Paths.get(home, "PandaBot");
+        boolean directoryExists = Files.exists(dirPath);
+
+        // create a dir for save files to be saved in
+        if (!directoryExists) {
+            try {
+                java.nio.file.Files.createDirectory(dirPath);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        Path saveFilePath = Paths.get(home, "PandaBot", fileName);
```

Comment:
Shadows line 14.

Reply:
Ah yes it did, thanks for pointing that out